src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlAttributes.cs

```diff
@@ -99,7 +83,7 @@ public XmlAttributes(ICustomAttributeProvider provider)
             XmlAnyElementAttribute? wildcard = null;
             for (int i = 0; i < attrs.Length; i++)
             {
-                if (attrs[i] is XmlIgnoreAttribute || attrs[i] is ObsoleteAttribute || attrs[i].GetType() == IgnoreAttribute)
+                if (attrs[i] is XmlIgnoreAttribute || attrs[i] is ObsoleteAttribute)
```

Comment:
Not sure why we used to have that IgnoreAttribute using reflection if we already had the check for XmlIgnoreAttribute here. Do you know why we had it in both places?

Reply:
It was originally to break a circular dependency between `mscorlib` (where AppDomainSetup lived) and `System` (where Xml serialization and `XmlIgnoreAttribute` lived). So they could ignore certain members on AppDomainSetup when it was serialized.